src/app/menu/menu.component.ts

```diff
@@ -49,4 +50,25 @@ export class MenuComponent implements OnInit {
             this._router.navigate(['/login']);
         });
     }
+
+    // Methods to add dynamic classes based on user\'s progress
+    calculateInnerClasses(stage: number, className: string) {
+      return  (this.checkStageStatus(stage)) ? {[className]: true, 'inner-circle': true} : {'unlocked-inner-circle': true, 'inner-circle': true};
+    }
+
+    calculateButtonClasses(stage: number, className: string) {
+      return  (this.checkStageStatus(stage)) ? {[className]: true, 'but': true} : {'unlocked-btn': true, 'but': true};
+    }
+
+    calculateButtonBackground(stage: number, name: string) {
+      return  (this.checkStageStatus(stage)) ? '../../assets/img/' + name + '.png' : '../../assets/img/unlocked.png';
+    }
+
+    calculateOuterClasses(stage: number, className: string) {
+      return  (this.checkStageStatus(stage)) ? {[className]: true, 'outer-circle': true} : {'unlocked-btn': true, 'outer-circle': true};
+    }
+
+    checkStageStatus(stage: number) {
```

Comment:
Add comments for the functions

Reply:
adding to it, the comments should have 
@params and @return explaining what it accepts and what it returns.